api/src/main/java/com/datastrato/gravitino/rel/SupportsPartitions.java

```diff
@@ -79,22 +80,49 @@ default boolean partitionExists(String partitionName) {
   /**
    * Drop a partition with specified name.
    *
-   * @param partitionName The identifier of the partition.
-   * @return true if a partition was deleted, false if the partition did not exist.
+   * @param partitionName the name of the partition
+   * @param ifExists If true, will not throw NoSuchPartitionException if the partition not exists
+   * @return true if a partition was deleted.
+   */
+  boolean dropPartition(String partitionName, boolean ifExists) throws NoSuchPartitionException;
+
+  /**
+   * Drop partitions with specified names.
+   *
+   * @param partitionNames the names of the partition
+   * @param ifExists If true, will not throw NoSuchPartitionException if the partition not exists
+   * @return true if all partitions was deleted.
    */
-  boolean dropPartition(String partitionName);
+  boolean dropPartitions(List<String> partitionNames, boolean ifExists)
```

Comment:
Since the `dropPartitions` interface is not implemented in this PR, I suggest removing it first in this PR

Reply:
I think that we can keep the api, and implement it in next version, just like addPartitions only support add one partition right now

Reply:
> just like addPartitions only support add one partition right now

There is not `addPartitions` method in API module

Reply:
OK , I'll remove it